core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -126,23 +126,9 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
   def merge(other: AccumulatorV2[IN, OUT]): Unit
 
   /**
-   * Access this accumulator's current value; only allowed on driver.
+   * Defines the current value of this accumulator
    */
-  final def value: OUT = {
-    if (atDriverSide) {
```

Comment:
We should put this check in the old accumulator, to avoid breaking its behaviour.